### PR TITLE
Add support for Honor 4X Qualcomm

### PR DESCRIPTION
**Pull request by Boom-Hacker**

### CODE REVIEW

File: dts/msm8916/msm8916-huawei-hx4qcom.dts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+#include <skeleton.dtsi>
+
+/ {
+	qcom,msm-id = <206 0>,<248 0>,<249 0>,<250 0>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8916/msm8916-huawei-hx4qcom.dts:7]: https://github.com/torvalds/linux/blob/efb448d0a3fca01bb987dd70963da6185b81751e/drivers/soc/qcom/socinfo.c#L110-L113

Keep only `<206 0>`(MSM8916/LTE) if there is no 3G/Wi-Fi only variant.

File: dts/msm8916/msm8916-huawei-hx4qcom.dts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	qcom,board-id = <8034 4>,<8050 4>,<8082 4>,<8098 4>,<8114 4>,<8115 4>,<8130 4>,<8131 4>;
+
+	model = "Huawei Honor 4X Qualcomm";
+	compatible = "huawei,h4xqcom", "qcom,msm8916", "lk2nd,device";
+
+	panel {
+		compatible = "huawei,h4xqcom";
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8916/msm8916-huawei-hx4qcom.dts:14]:
```suggestion
		compatible = "huawei,h4xqcom-panel";
```

or

```suggestion
		compatible = "huawei,cherry-panel";
```

Comment (Boom-Hacker) [reply dts/msm8916/msm8916-huawei-hx4qcom.dts:14]: many Honor 4X Kirin codename is cherry too,so i think use new codename is better to use "cherry"

Comment (Boom-Hacker) [reply dts/msm8916/msm8916-huawei-hx4qcom.dts:14]: btw,this is my first pull request(maybe),i will do what to continue pull request

File: README.md [hunk 3]
```diff
@@ -58,6 +58,7 @@ page on the EFIDroid wiki for an exact mapping of LK targets to SoCs.
 - GPLUS FL8005A
 - HTC One M8s - m8qlul (quirky - see comment in `dts/msm8916/msm8939-htc-m8qlul.dts`)
 - Huawei Ascend G7 - G7-L01
+- Huawei Honor 4X Qualcomm version and Huawei Ascend G620S - hx4qcom (cherry in stock)
```
Comment (wonderfulShrineMaidenOfParadise) [inline README.md:61]: Honor 4X or Ascend G620S? They are not the same phone.

File: target/msm8916/oem_panel.c [hunk 4]
```diff
@@ -138,6 +140,22 @@ static int init_panel_data(struct panel_struct *panelstruct,
 	int pan_type = PANEL_TYPE_DSI;
 
 	switch (panel_id) {
+	case CMI_NT35521_5P5_720P_VIDEO_PANEL:
```
Comment (stephan-gh) [inline target/msm8916/oem_panel.c:143]: The changes here in target/msm8916/oem_panel.c are not needed. It is enough to add it in lk2nd/panel.

File: dts/msm8916/msm8916-huawei-cherry.dts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+#include <skeleton.dtsi>
+
+/ {
+	qcom,msm-id = <206 0>;
+	qcom,board-id = <8034 4>,<8050 4>,<8082 4>,<8098 4>,<8114 4>,<8115 4>,<8130 4>,<8131 4>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8916/msm8916-huawei-cherry.dts:8]:
```suggestion
	qcom,board-id = <8034 4>, <8050 4>, <8082 4>, <8098 4>, <8114 4>,
			<8115 4>, <8130 4>, <8131 4>;
```

File: README.md [hunk 6]
```diff
@@ -58,6 +58,7 @@ page on the EFIDroid wiki for an exact mapping of LK targets to SoCs.
 - GPLUS FL8005A
 - HTC One M8s - m8qlul (quirky - see comment in `dts/msm8916/msm8939-htc-m8qlul.dts`)
 - Huawei Ascend G7 - G7-L01
+- Huawei Honor 4X Qualcomm - hx4qcom (cherry in stock)
```
Comment (wonderfulShrineMaidenOfParadise) [inline README.md:61]:
```suggestion
- Huawei Honor 4X Qualcomm - cherry
```

**Review by wonderfulShrineMaidenOfParadise — CHANGES_REQUESTED** [review]

**Pull request closed without merging**